src/pipx/package_specifier.py

```diff
@@ -161,7 +161,27 @@ def parse_specifier_for_install(package_spec: str, pip_args: List[str]) -> Tuple
         )
         pip_args.remove("--editable")
 
-    return (package_or_url, pip_args)
+    for index, option in enumerate(pip_args):
+        if not option.startswith(("-c", "--constraint")):
+            continue
+
+        if option == "-c" or option == "--constraint":
+            next_option_index = index + 1
+            if next_option_index < len(pip_args):
+                next_option = pip_args[next_option_index]
```

Comment:
Let's rename this and also `next_option_index`.
```suggestion
                constraints_file = pip_args[argument_index]
```

Reply:
done